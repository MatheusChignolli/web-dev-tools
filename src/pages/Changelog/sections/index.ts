export { default as Release } from './Release'
