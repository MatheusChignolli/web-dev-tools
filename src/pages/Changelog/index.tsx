import { useState } from 'react'
import { useQuery } from 'react-query'
import { dateUtils } from '~utils'
import { Grid, Loader } from '~components'
import { useTheme } from '~hooks'
import { githubServices } from '~services'
import { Release } from './sections'

const referencePath = 'version-accordion-'

const Changelog = () => {
  const theme = useTheme()
  const [accordionReference, setAccordionReference] = useState(`${referencePath}0`)

  const { data, isLoading } = useQuery(
    'getReleasesFromRepo',
    () => githubServices.getReleasesFromRepo(githubServices.repoConfig),
    {
      refetchOnWindowFocus: false,
      onError(error) {
        console.error('getReleasesFromRepo: ', error)
      },
    }
  )

  if (isLoading) {
    return <Loader />
  }

  if (data?.length === 0) {
    return <h1>Sem conte√∫do</h1>
  }

  return (
    <Grid container padding={`${theme.spacing(2)} 0px`} direction="column">
      {data?.map(({ body, created_at, tag_name }, index) => {
        const reference = `${referencePath}${index}`
        return (
          <Release
            key={`key-${reference}`}
            body={body}
            title={`[${tag_name}] - ${dateUtils.format(created_at)}`}
            expanded={accordionReference === reference}
            onChange={() => {
              setAccordionReference((prevState) =>
                prevState === reference ? '' : reference
              )
            }}
          />
        )
      })}
    </Grid>
  )
}

export default Changelog
