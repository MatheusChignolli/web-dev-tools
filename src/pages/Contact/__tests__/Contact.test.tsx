import { screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { render } from '~tests'
import Changelog from '../'

describe('Contact', () => {
  it('should create a link to mail message to app`s author', async () => {
    const { container } = render(<Changelog />)

    await userEvent.type(
      screen.getByRole('textbox', { name: 'E-mail' }),
      'example@example.com'
    )
    await userEvent.type(screen.getByRole('textbox', { name: 'Assunto' }), 'Subject')
    await userEvent.type(
      screen.getByRole('textbox', { name: 'Mensagem' }),
      'Message'
    )

    await userEvent.click(
      screen.getByRole('button', {
        name: 'Botão para submeter formulário de contato',
      })
    )

    expect(container).toMatchSnapshot()
  })
})
