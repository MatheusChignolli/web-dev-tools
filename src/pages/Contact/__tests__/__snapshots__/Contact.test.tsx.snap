// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact should create a link to mail message to app\`s author 1`] = `
<div>
  <div
    class="MuiStack-root css-im9u24-MuiStack-root"
  >
    <h3
      class="MuiTypography-root MuiTypography-h3 css-129yi13-MuiTypography-root"
    >
      Contato
       
      <div
        class="MuiStack-root css-1f615ih-MuiStack-root"
      >
        <a
          aria-label="Link para o repositório da aplicação"
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone sc-eqUAAy fFmULB css-k6ij58-MuiTypography-root-MuiLink-root"
          href="https://github.com/MatheusChignolli/web-dev-tools"
          target="_blank"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-ft5hsy-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              aria-label="Ícone do GitHub arredondado e sem bordas"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1jxdcj3-MuiSvgIcon-root"
              data-testid="GitHubIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </a>
        <a
          aria-label="Link para o LinkedIn do autor da aplicação"
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone sc-eqUAAy fFmULB css-k6ij58-MuiTypography-root-MuiLink-root"
          href="https://www.linkedin.com/in/matheus-chignolli-a0115b155/"
          target="_blank"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-ft5hsy-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              aria-label="Ícone do LinkedIn arredondado e sem bordas"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1jxdcj3-MuiSvgIcon-root"
              data-testid="LinkedInIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </a>
      </div>
    </h3>
    <p
      class="MuiTypography-root MuiTypography-body1 css-ukks2v-MuiTypography-root"
    >
      Valorizamos seu feedback, perguntas e sugestões. Se você precisa de suporte, tem uma dúvida sobre nossas ferramentas ou deseja compartilhar suas ideias, estamos aqui para ajudar. Por favor, utilize as informações de contato abaixo para entrar em contato conosco.
    </p>
    <form>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <div
            aria-label="Campo para digitar o seu e-mail"
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-2dhdve-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1aacg4d-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for="email"
              id="email-label"
            >
              E-mail
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1gpgxrm-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="email"
                name="email"
                placeholder="seu-email@exemplo.com.br"
                required=""
                type="email"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-14lo706"
                >
                  <span>
                    E-mail
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <div
            aria-label="Campo para digitar o assunto da sua mensagem"
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-2dhdve-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1aacg4d-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for="subject"
              id="subject-label"
            >
              Assunto
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1gpgxrm-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="subject"
                name="subject"
                placeholder="Feedback sobre a funcionalidade..."
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-14lo706"
                >
                  <span>
                    Assunto
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            aria-label="Campo para digitar a sua mensagem"
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-2dhdve-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1aacg4d-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for="message"
              id="message-label"
            >
              Mensagem
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-h4k37g-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <textarea
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                id="message"
                name="message"
                placeholder="A funciondalide de geração de documentos..."
                required=""
                rows="4"
                style="height: 0px; overflow: hidden;"
              />
              <textarea
                aria-hidden="true"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                readonly=""
                style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px; width: 100%;"
                tabindex="-1"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-14lo706"
                >
                  <span>
                    Mensagem
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <button
            aria-label="Botão para submeter formulário de contato"
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-1t3g34h-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            Enviar
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            >
              <span
                class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
                style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
              >
                <span
                  class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
                />
              </span>
            </span>
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
